extensions/servlet/src/com/google/inject/servlet/ServletScopes.java

```diff
@@ -218,9 +220,15 @@ public String toString() {
    *     thread, or if the request has completed.
    * 
    * @since 3.0
+   * @deprecated You probably want to use {@code transferRequest} instead
```

Comment:
We do something similar, with a custom `@ThreadSafeRequestScoped` scope.  We put those objects in the `seedMap` to share them, and rely on `continueRequest()` to re-create new instances of the non-thread-safe `@RequestScoped` objects in the other threads.  I agree that `continueRequest()` on its own isn't perfect.

Well, as long as deprecated doesn't mean "getting deleted soon" I'll keep using it :).